common/chat.ts

```diff
@@ -260,13 +262,22 @@ export interface ChatUserProfile {
 
 export type ChatUserProfileJson = Jsonify<ChatUserProfile>
 
+export function toChatUserProfileJson(chatUserProfile: ChatUserProfile): ChatUserProfileJson {
+  return {
+    userId: chatUserProfile.userId,
+    channelName: chatUserProfile.channelName,
+    joinDate: Number(chatUserProfile.joinDate),
+    isModerator: chatUserProfile.isModerator,
+  }
+}
+
 /**
  * The response returned when fetching a profile of a user in a specific chat channel. The profile
- * itself can be `null` in case the user has left the channel, but their name is still visible in
- * old messages.
+ * itself can be `undefined` in case the user has left the channel, but their name is still visible
```

Comment:
nit: If you have a comment about a particular field, it should probably just be a doc comment on that field